dbms/src/Client/ConnectionPoolWithFailover.h

```diff
@@ -34,7 +37,8 @@ class ConnectionPoolWithFailover : public IConnectionPool, private PoolWithFailo
     ConnectionPoolWithFailover(
             ConnectionPoolPtrs nested_pools_,
             LoadBalancing load_balancing,
-            time_t decrease_error_period_ = DBMS_CONNECTION_POOL_WITH_FAILOVER_DEFAULT_DECREASE_ERROR_PERIOD);
+            time_t decrease_error_period_ = DBMS_CONNECTION_POOL_WITH_FAILOVER_DEFAULT_DECREASE_ERROR_PERIOD,
+            size_t max_error_cap = DBMS_CONNECTION_POOL_WITH_FAILOVER_MAX_ERROR_COUNT);
```

Comment:
Does this mean that some callers use the pool with compile-time settings, and some -- with user-specified settings? What is the reason for that?

Reply:
`ClickHouseDictionarySource` uses defaults, the main reason is that these settings are for controlling distributed tables / replicas even the setting name begins with `replica_*`. I am not sure if error cap/recovery rate of remote dictionaries should be configurable, but it is definitely a bad idea to use same settings for both dictionaries and cluster replicas.

Reply:
OK, I see. I'd make the parameters explicit so that the caller doesn't forget to set the right values, but that's up to you.

There is another use in StorageDistributedDirectoryMonitor::createPool that has default timeouts. Not sure what it's used for, is this OK?

Reply:
Fixed that, also updated `replica_error_decrease_period` to be `SettingSeconds` instead of `SettingUInt64`.